Server/Controllers/ServerEventsController.cs

```diff
@@ -43,33 +45,57 @@ IEnumerable<IFinalizedProvider> eventFinalizers
-        public void SSE()
+        public async Task SSE()
         {
             // Set the response headers; this tells the client we're initiating SSE
             Response.Headers.ContentType = "text/event-stream";
             Response.Headers.CacheControl = "no-cache";
-            Response.Headers.Connection = "keep-alive";
+            Response.Headers.Append("Connection", "keep-alive");
+
+            TaskCompletionSource tcs = new();
+            CancellationToken cancellationToken = HttpContext.RequestAborted;
 
             foreach (IFinalizedProvider eventFinalizer in _eventFinalizers)
             {
                 // Subscribe to finalizers
                 eventFinalizer.OnDataProvided += async payload =>
                 {
-                    // Leaving these here just in case...
-                    //_logger.LogInformation("Sending event provided by {evtFinalizerType}.", eventFinalizer.GetType().Name);
-                    //_logger.LogDebug("Tag: {tag}\nContent: {content}", payload.Data.tag, payload.Data.content);
+                    try
+                    {
+                        // Leaving these here just in case...
+                        //_logger.LogInformation("Sending event provided by {evtFinalizerType}.", eventFinalizer.GetType().Name);
+                        _logger.LogDebug(
+                            "Tag: {tag}\nContent: {content}",
+                            payload.Data.tag,
+                            payload.Data.content
+                        );
 
-                    // Send the tagged event in a properly formatted way
-                    await Response.WriteAsync($"event: {payload.Data.tag}\n");
-                    await Response.WriteAsync($"data: ");
-                    // Convert the content to JSON
-                    await Response.WriteJSONAsync(payload.Data.content);
-                    await Response.WriteAsync("@");
-                    await Response.WriteJSONAsync(payload.DataStamp);
-                    await Response.WriteAsync("\n\n");
-                    await Response.Body.FlushAsync();
+                        // Send the tagged event in a properly formatted way
+                        await Response.WriteAsync($"event: {payload.Data.tag}\n");
+                        await Response.WriteAsync($"data: ");
+                        // Convert the content to JSON
+                        await Response.WriteJSONAsync(payload.Data.content);
+                        await Response.WriteAsync("@");
+                        await Response.WriteJSONAsync(payload.DataStamp);
+                        await Response.WriteAsync("\n\n");
+                        await Response.Body.FlushAsync();
+                    }
+                    catch (Exception e)
+                    {
+                        _logger.LogError(
+                            e,
+                            "Error in SSE connection. The client may have disconnected."
+                        );
```

Comment:
Since this doesn't *crash* the server - and is to some extent normal behaviour, I think logging it as an error might be too harsh. Also, since it's normal, it shouldn't be this verbose.
```suggestion
                        _logger.LogWarning(
                            "Error in SSE connection. The client may have disconnected."
                        );
```
Also, maybe only the exception type that comes from a client disconnecting should be caught.